From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zhangyuheng <zhangyuheng@lunadeer.cn>
Date: Thu, 22 May 2025 13:17:17 +0800
Subject: [PATCH] DeerFoliaPlus configuration


diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 527aa58ca531b2eec6b762b1f24ed7783179efc9..3da8e9f9b0cc5c8e677322d12d1371689640f413 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -241,6 +241,9 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         // DeerFolia start - initialize config
         cn.lunadeer.mc.deerfolia.DeerFoliaConfiguration.load();
         // DeerFolia end - initialize config
+        // DeerFoliaPlus start - initialize config
+        cn.lunadeer.mc.deerfoliaplus.configurations.DeerFoliaPlusConfiguration.load();
+        // DeerFoliaPlus end - initialize config
         this.server.spark.enableEarlyIfRequested(); // Paper - spark
         // Paper start - fix converting txt to json file; convert old users earlier after PlayerList creation but before file load/save
         if (this.convertOldUsers()) {
