From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zhangyuheng <zhangyuheng@lunadeer.cn>
Date: Thu, 22 May 2025 13:17:17 +0800
Subject: [PATCH] DeerFoliaPlus configuration


diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index d550b2d58de031eb470a9452af7e0f37caf739b4..741e4ee63020665f07017d21111095e53325ead7 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -264,6 +264,9 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         // DeerFolia start - initialize config
         cn.lunadeer.mc.deerfolia.DeerFoliaConfiguration.load();
         // DeerFolia end - initialize config
+        // DeerFoliaPlus start - initialize config
+        cn.lunadeer.mc.deerfoliaplus.configurations.DeerFoliaPlusConfiguration.load();
+        // DeerFoliaPlus end - initialize config
         this.server.spark.enableEarlyIfRequested(); // Paper - spark
         // Paper start - fix converting txt to json file; convert old users earlier after PlayerList creation but before file load/save
         if (this.convertOldUsers()) {
