From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zhangyuheng <zhangyuheng@lunadeer.cn>
Date: Tue, 3 Mar 2026 10:04:23 +0800
Subject: [PATCH] Bedrock-style stronghold generation


diff --git a/net/minecraft/world/level/chunk/ChunkGeneratorStructureState.java b/net/minecraft/world/level/chunk/ChunkGeneratorStructureState.java
index 8b4ea2b16da45bcefe743990866e43c94e121825..d39e3814f7c03d3c1e13b984c9a8f5005668fbdd 100644
--- a/net/minecraft/world/level/chunk/ChunkGeneratorStructureState.java
+++ b/net/minecraft/world/level/chunk/ChunkGeneratorStructureState.java
@@ -73,6 +73,21 @@ public class ChunkGeneratorStructureState {
         return list.stream().map((holder) -> {
             StructureSet structureset = holder.value();
             final Holder<StructureSet> newHolder; // Paper - Add missing structure set seed configs
+            // DeerFoliaPlus start - Bedrock-style stronghold generation
+            if (structureset.placement() instanceof ConcentricRingsStructurePlacement concentricConfig
+                    && holder.unwrapKey().isPresent()
+                    && holder.unwrapKey().get().equals(net.minecraft.world.level.levelgen.structure.BuiltinStructureSets.STRONGHOLDS)
+                    && cn.lunadeer.mc.deerfoliaplus.configurations.DeerFoliaPlusConfiguration.bedrockStrongholdGeneration.enabled) {
+                structureset = new StructureSet(structureset.structures(),
+                        new net.minecraft.world.level.levelgen.structure.placement.RandomSpreadStructurePlacement(
+                                cn.lunadeer.mc.deerfoliaplus.configurations.DeerFoliaPlusConfiguration.bedrockStrongholdGeneration.spacing,
+                                cn.lunadeer.mc.deerfoliaplus.configurations.DeerFoliaPlusConfiguration.bedrockStrongholdGeneration.separation,
+                                net.minecraft.world.level.levelgen.structure.placement.RandomSpreadType.LINEAR,
+                                10387319 // DeerFoliaPlus - unique salt for random stronghold placement
+                        ));
+                newHolder = Holder.direct(structureset);
+            } else
+            // DeerFoliaPlus end - Bedrock-style stronghold generation
             if (structureset.placement() instanceof net.minecraft.world.level.levelgen.structure.placement.RandomSpreadStructurePlacement randomConfig && holder.unwrapKey().orElseThrow().identifier().getNamespace().equals(net.minecraft.resources.Identifier.DEFAULT_NAMESPACE)) { // Paper - Add missing structure set seed configs; check namespace cause datapacks could add structure sets with the same path
                 String name = holder.unwrapKey().orElseThrow().identifier().getPath();
                 int seed = randomConfig.salt;
