From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zhangyuheng <zhangyuheng@lunadeer.cn>
Date: Thu, 22 May 2025 15:33:22 +0800
Subject: [PATCH] Leaves fake player


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 325d797766de527268eb18bcf9db594ca3a0c519..7a91793d26d74d9530ffec607a1dcf4c31bd09cf 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -3045,4 +3045,16 @@ public final class Bukkit {
     public static void restart() {
         server.restart();
     }
+
+
+    // Leaves start - Bot API
+    /**
+     * Returns a bot manager.
+     *
+     * @return Bot Manager
+     */
+    public static @NotNull org.leavesmc.leaves.entity.BotManager getBotManager() {
+        return server.getBotManager();
+    }
+    // Leaves end - Bot API
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 067d901c69a54c4b6ae5a00488f0dacc5fde7961..008577e5d711f796b23acdfcb194c21ff25c09c7 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2759,6 +2759,15 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     public boolean isGlobalTickThread();
     // Paper end - Folia region threading API
 
+    // Leaves start - Bot API
+    /**
+     * Returns a bot manager.
+     *
+     * @return Bot Manager
+     */
+    @NotNull org.leavesmc.leaves.entity.BotManager getBotManager();
+    // Leaves end - Bot API
+
     // Paper start - API to check if the server is sleeping
     /**
      * Returns whether the server is sleeping/paused.
