From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zhangyuheng <zhangyuheng@lunadeer.cn>
Date: Thu, 22 May 2025 15:33:22 +0800
Subject: [PATCH] Leaves fake player


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 64694483664c5a7380b1ce7846f2eef59004762f..1c8527e4fcdfd53393d6e25f77d217215ff75a75 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -3032,4 +3032,16 @@ public final class Bukkit {
     public static void restart() {
         server.restart();
     }
+
+
+    // Leaves start - Bot API
+    /**
+     * Returns a bot manager.
+     *
+     * @return Bot Manager
+     */
+    public static @NotNull org.leavesmc.leaves.entity.BotManager getBotManager() {
+        return server.getBotManager();
+    }
+    // Leaves end - Bot API
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index a0922805fbee8e064a74115fbf93e9e8a98772cb..787b0c0ad631997a8a62e3cafd15d722f8c2376d 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2683,6 +2683,15 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     public boolean isGlobalTickThread();
     // Paper end - Folia region threading API
 
+    // Leaves start - Bot API
+    /**
+     * Returns a bot manager.
+     *
+     * @return Bot Manager
+     */
+    @NotNull org.leavesmc.leaves.entity.BotManager getBotManager();
+    // Leaves end - Bot API
+
     // Paper start - API to check if the server is sleeping
     /**
      * Returns whether the server is sleeping/paused.
