From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zhangyuheng <zhangyuheng@lunadeer.cn>
Date: Thu, 22 May 2025 15:33:22 +0800
Subject: [PATCH] Leaves fake player


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index b9d34c0b80a5d0c26f43e4959c3a8256a854585a..9431a9938d2aaa362ba6dcdec2eefe28da503a81 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -3039,4 +3039,16 @@ public final class Bukkit {
     public static void restart() {
         server.restart();
     }
+
+
+    // Leaves start - Bot API
+    /**
+     * Returns a bot manager.
+     *
+     * @return Bot Manager
+     */
+    public static @NotNull org.leavesmc.leaves.entity.BotManager getBotManager() {
+        return server.getBotManager();
+    }
+    // Leaves end - Bot API
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 1dc5d8c96786950fc98cada5bad981c0c716b154..28e235436ec733e25e76dd9de65da9670426a0ea 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2746,6 +2746,15 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     public boolean isGlobalTickThread();
     // Paper end - Folia region threading API
 
+    // Leaves start - Bot API
+    /**
+     * Returns a bot manager.
+     *
+     * @return Bot Manager
+     */
+    @NotNull org.leavesmc.leaves.entity.BotManager getBotManager();
+    // Leaves end - Bot API
+
     // Paper start - API to check if the server is sleeping
     /**
      * Returns whether the server is sleeping/paused.
